_episodes/05-pulling_pushing.md

```diff
@@ -0,0 +1,146 @@
+
+As a reminder, remote and local repositories are not automatically synchronised, but
+rather it is a manual process done via `git pull` and `git push` commands. This
+synchronisation needs to be done **branch by branch** with all of those you want to keep
+in sync.
+
+### Pushing
+
+* Its basic use is to synchronise **any committed changes** in your current
+ branch to its upstream branch: `git push`.
+* Changes in the staging area will not be synchronised.
+![Git collaborative]({{ site.baseurl }}/fig/push.png "Push a branch
+."){:class="img-responsive"}
+* If the current branch has no upstream yet, you can configure one by doing
+`git push -u origin BRANCH_NAME`, as done with `main` in the exercise
+ above.
+![Git collaborative]({{ site.baseurl }}/fig/push_u.png "Push a branch without
+ upstream yet."){:class="img-responsive"}
+* `push` only operates on your current branch. If you want to push another
+ branch, you have to `checkout` that branch first.
+* If the upstream branch has changes you do not have in the local branch, the
+ command will fail, requesting you to pull those changes first.
+
+Let's try to push changes to the `main` branch. First make sure you are on the `main` branch.
+
+```sh
+git switch main
+```
+{: .commands}
+```
+Switched to branch 'main'
+Your branch is ahead of 'origin/main' by 6 commits.
+  (use "git push" to publish your local commits)
+```
+{: .output}
+
+Notice that git is suggesting you to use `git push` to publish your local commits. Let's do that:
+
+```sh
+git push
+```
+{: .commands}
+```
+Enumerating objects: 21, done.
+Counting objects: 100% (21/21), done.
+Delta compression using up to 8 threads
+Compressing objects: 100% (18/18), done.
+Writing objects: 100% (18/18), 1.83 KiB | 938.00 KiB/s, done.
+Total 18 (delta 4), reused 0 (delta 0), pack-reused 0 (from 0)
+remote: Resolving deltas: 100% (4/4), completed with 1 local object.
+To https://github.com/username/recipe.git
+   57d4505..d10e1e9  main -> main
+```
+{: .output}
+
+> ## Now you try
+>
+> You should now be on the `main` branch. Try switching to the `spicy`
+> branch and pushing changes to it.
+>
+> 1. Check the current branch using `git branch`.
+> 1. If the current branch is `main`, then switch to `spicy` using `git switch spicy`.
+> 1. Push changes to the `spicy` branch using `git push`.
+>
+> > ## Solution
+> >
+> > ```
+> > git branch
+> > git switch spicy
+> > git push
+> > ```
+> > {: .commands}
+> > This should give rise to an error that the current branch `spicy` has no upstream branch:
+> > ```
+> > fatal: The current branch spicy has no upstream branch.
+> > To push the current branch and set the remote as upstream, use
+> >
+> >     git push --set-upstream origin spicy
+> >
+> > To have this happen automatically for branches without a tracking
+> > upstream, see 'push.autoSetupRemote' in 'git help config'.
+> > ```
+> > Push again by setting the upstream:
+> > ```
+> > git push --set-upstream origin spicy
```

Comment:
I think earlier in the text we tell students to use the `-u` flag instead. Maybe it would be worth mentioning the `--set-upstream` flag when they first encounter it and telling them that `-u` is a shortcut for it?

Reply:
Yes, I agree. Put this one earlier and mention `-u` can be used instead